### PR TITLE
[coqtop] fix directory for subdirectory targets.

### PR DESCRIPTION
**Pull request by Alizter**
Subsumes #5868

Close #5868
Fix #5552

### CODE REVIEW

File: test/blackbox-tests/test-cases/coq/coqtop-nested.t/run.t [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  $ dune build --display=short theories/c.vo
+        coqdep theories/c.v.d
+        coqdep theories/d.v.d
+          coqc theories/.d.aux,theories/d.{glob,vo}
+          coqc theories/.c.aux,theories/c.{glob,vo}
+  $ dune build --display=short theories/b/b.vo
+        coqdep theories/b/b.v.d
+        coqdep theories/a/a.v.d
+          coqc theories/a/.a.aux,theories/a/a.{glob,vo}
+          coqc theories/b/.b.aux,theories/b/b.{glob,vo}
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/coq/coqtop-nested.t/run.t:12]: could you remove the `--display=short` from the tests.

File: test/blackbox-tests/test-cases/coq/coqtop-root.t/run.t [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  $ dune coq top --toplevel=echo -- foo.v
+  Warning: No dune-project file has been found. A default one is assumed but
+  the project might break when dune is upgraded. Please create a dune-project
+  file.
+  Hint: generate the project file with: $ dune init project <name>
+  File "dune", line 1, characters 0-24:
+  1 | (coq.theory
+  2 |  (name foo))
+  Error: 'coq.theory' is available only when coq is enabled in the dune-project
+  file. You must enable it using (using coq 0.4) in your dune-project file.
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/coq/coqtop-root.t/run.t:15]: you should leave a comment that this case is broken now b/c of `--root`.

File: CHANGES.md [hunk 3]
```diff
@@ -13,6 +13,10 @@
 3.3.0 (17-06-2022)
 ------------------
 
+- Fix ``dune coq top`` when it is invoked on files from a subdirectory of the
```
Comment (rgrinberg) [inline CHANGES.md:16]: This should use markdown syntax. So a single backtick.

**Comment by ejgallego** [comment]
This still seems broken for Coq's stdlib from the Coq repos, I'll open an issue.

**Pull request merged**